app/models/miq_dialog.rb

```diff
@@ -1,30 +1,38 @@
 class MiqDialog < ApplicationRecord
-  validates_presence_of   :name, :description
-  validates_uniqueness_of :name, :scope => :dialog_type, :case_sensitive => false
+  validates :name, :description, :presence => true
+  validates :name, :uniqueness => { :scope => :dialog_type, :case_sensitive => false }
 
-  DIALOG_DIR = Rails.root.join("product/dialogs/miq_dialogs")
+  DIALOG_DIR = "product/dialogs/miq_dialogs".freeze
 
   DIALOG_TYPES = [
     [_("VM Provision"),                "MiqProvisionWorkflow"],
     [_("Configured System Provision"), "MiqProvisionConfiguredSystemWorkflow"],
     [_("Host Provision"),              "MiqHostProvisionWorkflow"],
     [_("VM Migrate"),                  "VmMigrateWorkflow"],
-  ]
+  ].freeze
 
   serialize :content
 
   def self.seed
-    sync_from_dir
+    sync_from_dir(Rails.root)
+    sync_from_plugins
   end
 
-  def self.sync_from_dir
-    Dir.glob(File.join(DIALOG_DIR, "*.yaml")).each { |f| sync_from_file(f) }
+  def self.sync_from_dir(root)
+    root = root.join(DIALOG_DIR)
```

Comment:
Also, in provider plugins, I would expect dialogs to live in `content/dialogs` to match the pattern we started in the `manageiq-content` repo.